packages/affiliates/libs/contracts.js

```diff
@@ -27,11 +28,76 @@ function decodeAttribution(
   data,
   delimiter = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000000000000000000000"
 ) {
+  assert(data, "requires data to decode");
   return data.split(delimiter)[1];
 }
 
+// Just wraps abi to pass through to contract
+// Lookup by erc20 address
+function Erc20({ abi = getAbi("ERC20"), web3 }) {
+  assert(abi, "requires abi for erc20");
+  const contract = new web3.eth.Contract(abi);
+  function decimals(address) {
+    assert(address, "requires address");
+    contract.options.address = address;
+    return contract.methods.decimals().call();
+  }
+  return {
+    decimals
+  };
+}
+
+// Wrapper for some basic emp functionality.
+// Currently we just need token and collateral info
```

Comment:
nit: you can make these comments longer. our styling guide prescribes 120 char line length so this does not need to be 3 lines